opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -0,0 +1,999 @@
+        if (CUDA_SUCCESS != result) {
+            opal_output_verbose(5, opal_accelerator_base_framework.framework_output,
+                                "CUDA: cuMemGetAddressRange failed on this pointer: result=%d, buf=%p "
+                                "Overriding check and setting to host pointer. ",
+                                result, (void *) dbuf);
+            /* This cannot be GPU memory if the previous call failed */
+            return 0;
+        }
+    }
+
+    return 1;
+}
+
+static int accelerator_cuda_create_stream(opal_accelerator_stream_t *stream)
+{
+    CUresult result;
+
+    if (NULL == stream) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    stream->stream = malloc(sizeof(CUstream));
+    if (NULL == stream->stream) {
+        return OPAL_ERR_OUT_OF_RESOURCE;
+    }
+
+    OPAL_THREAD_LOCK(&opal_accelerator_cuda_stream_lock);
+    result = opal_accelerator_cuda_func.cuStreamCreate(stream->stream, 0);
+    OPAL_THREAD_UNLOCK(&opal_accelerator_cuda_stream_lock);
+    if (OPAL_UNLIKELY(result != CUDA_SUCCESS)) {
+        opal_show_help("help-accelerator-cuda.txt", "cuStreamCreate failed", true,
+                       OPAL_PROC_MY_HOSTNAME, result);
+        free(stream->stream);
+        return OPAL_ERROR;
+    }
+    return OPAL_SUCCESS;
+}
+
+static int accelerator_cuda_destroy_stream(opal_accelerator_stream_t *stream)
+{
+    CUresult result;
+    if (NULL != stream) {
+        if (NULL != stream->stream) {
+            OPAL_THREAD_LOCK(&opal_accelerator_cuda_stream_lock);
+            result = opal_accelerator_cuda_func.cuStreamDestroy(*(CUstream *)stream->stream);
+            OPAL_THREAD_UNLOCK(&opal_accelerator_cuda_stream_lock);
+            if (OPAL_UNLIKELY(CUDA_SUCCESS != result)) {
+                opal_show_help("help-accelerator-cuda.txt", "cuStreamDestroy failed", true,
+                               result);
+                return OPAL_ERROR;
+            }
+            free(stream->stream);
+        }
+    }
+    return OPAL_SUCCESS;
+}
+
+static int accelerator_cuda_synchronize_stream(opal_accelerator_stream_t *stream)
+{
+    CUresult result;
+
+    if (NULL == stream) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    OPAL_THREAD_LOCK(&opal_accelerator_cuda_stream_lock);
+    result = opal_accelerator_cuda_func.cuStreamSynchronize(*(CUstream *)stream->stream);
+    OPAL_THREAD_UNLOCK(&opal_accelerator_cuda_stream_lock);
+    if (OPAL_UNLIKELY(CUDA_SUCCESS != result)) {
+        opal_show_help("help-accelerator-cuda.txt", "cuStreamSynchronize failed", true,
+                       OPAL_PROC_MY_HOSTNAME, result);
+        return OPAL_ERROR;
+    }
+    return OPAL_SUCCESS;
+}
+
+static int accelerator_cuda_create_event(opal_accelerator_event_t *event)
+{
+    CUresult result;
+
+    if (NULL == event) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    event->event = malloc(sizeof(CUevent));
+    if (NULL == event->event) {
+        free(event);
```

Comment:
I don't think the event should be freed here.

Reply:
Yeah this is a bug due to an API change